File: python-examples/browser-sdk-showcase/api/save-file-to-s3.py
# https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/save_file_to_s3
from typing import TypedDict

from intuned_browser import go_to_url, save_file_to_s3
from playwright.async_api import Page


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params | None = None, **_kwargs):
    await go_to_url(page, "https://sandbox.intuned.dev/pdfs")
    # Download using a locator
    uploaded_file = await save_file_to_s3(
        page=page,
        trigger=page.locator("xpath=//tbody/tr[1]//*[name()='svg']"),
    )
    signed_url = await uploaded_file.get_signed_url()
    print(f"Signed URL: {signed_url}")
    return {
        "file_name": uploaded_file.file_name,
        "signed_url": signed_url,
        "message": "File uploaded successfully to S3"
    }

File: typescript-examples/browser-sdk-showcase/api/save-file-to-s3.ts
// https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/saveFileToS3
import { BrowserContext, Page } from "playwright";
import { saveFileToS3, goToUrl } from "@intuned/browser";

interface Params {
  // No params needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  await goToUrl({ page, url: "https://sandbox.intuned.dev/pdfs" });

  // Download using a locator
  const uploadedFile = await saveFileToS3({
    page,
    trigger: page.locator("xpath=//tbody/tr[1]//*[name()='svg']"),
  });

  const signedUrl = await uploadedFile.getSignedUrl();
  console.log(`Signed URL: ${signedUrl}`);
  return signedUrl;
}

